deliveries/decentralized_oracles-milestone_2.md

```diff
@@ -0,0 +1,53 @@
+
+Our team has applied to the Phala builder program to develop the offchain rollup request-response for Ink! smart contracts.
+
+When this milestone is completed, we would like to use this offchain rollup request-response model to build:
+- VRF Oracle. It should generate cryptographically secure random numbers for Ink! smart contracts
+- Graph API Oracle that:
+    - accesses SubQuery/SubSquid via an URI
+    - transform and format the response via the JS engine developed by the Phala team
+    - send the output to the Ink! smart contract
+
+In the milestone 2, concerning the Graph Api Oracle, we provide:
+- the Phala offchain rollup for the Graph Api Oracle 
+- the Ink! smart contract to interact with the offchain rollup
+- a UI to test and play with these smart contracts.
+- A SubQuery project that indexes the data from dApp Staking on Astar Network
+- the documentation to use these contracts
+
+**Deliverables**
+> Please provide a list of all deliverables of the milestone extracted from the initial application and a link to the deliverable itself. Ideally all links inside the below table should include a commit hash, which will be used for testing. If you don't provide a commit hash, we will work off the default branch of your repository. Thus, if you plan on continuing work after delivery, we suggest you create a separate branch for either the delivery or your continuing work.
+> If there is anything particular about any of the deliverables we or a future reader should know, use the respective `Notes` column.
+
+Demo here: https://graph-api-decentralized-oracle.substrate.fi
+
+| Number | Deliverable                             | Link                                                                              | Notes                                                                  |
+|--------|-----------------------------------------|-----------------------------------------------------------------------------------|------------------------------------------------------------------------|
+| 1.     | Phat contract: JS Offchain Rollup       | https://github.com/decentralized-oracles/graph-api-oracle/tree/main/phat/contracts/js_offchain_rollup|                                                                        | 
+| 2.     | Ink! smart contract: graph api consumer | https://github.com/decentralized-oracles/graph-api-oracle/tree/main/ink/contracts/graph_api_consumer|                                                                        | 
+| 3.     | Documentation                           | https://github.com/decentralized-oracles/graph-api-oracle                                |                                                                        | 
+| 4.     | UI                                      | https://github.com/decentralized-oracles/graph-api-ui                            | Demo here: https://graph-api-decentralized-oracle.substrate.fi         |  
```

Comment:
There's a lot of unused variables and duplicate import statements. See the error logs in the terminal.

[Here](https://github.com/decentralized-oracles/graph-api-ui/blob/8dc98370647fd003186f2c2bc7d662e28a6362df/src/components/ErrorContract.js#L1) is one

Reply:
Thanks for spotting these issues. We will fix them.

Reply:
> There's a lot of unused variables and duplicate import statements. See the error logs in the terminal.
> 
> [Here](https://github.com/decentralized-oracles/graph-api-ui/blob/8dc98370647fd003186f2c2bc7d662e28a6362df/src/components/ErrorContract.js#L1) is one

thanks for reporting
I've pushed a commit with improvements
https://github.com/decentralized-oracles/graph-api-ui/commit/117bf41adaa9e3e0b19f2978ca108735c8a9cd83

console is indeed much more clean now